lib/apply-extends.ts

```diff
@@ -32,7 +32,7 @@ function mergeDeep (config1: Dictionary, config2: Dictionary) {
   return target
 }
 
-export function applyExtends (config: Dictionary, cwd: string, mergeExtends: boolean) {
+export function applyExtends (config: Dictionary, cwd: string, mergeExtends?: boolean) {
```

Comment:
`applyExtends` does not only handle `Configuration` objects (yargs-parser and yargs config), but also more generic app config. So using `Configuration` would be too restrictive here.

Reply:
> I think mergeExtends would be cleaner by setting a default value of false rather than marking it as an optional parameter.
> A return type would be nice too.

Committed